src/hooks/useListEventsFiltered.ts

```diff
@@ -50,6 +51,18 @@ export default function useListEventsFiltered(
       })
     }
 
+    if (filter.tag && filter.tag != "all") {
+      const matches = filter.tag.toLowerCase()
+
+      events = events.filter((event) => {
+        const categoryList = new Set<string>()
```

Comment:
you can use:

```suggestion
        const categoryList = new Set<string>(event.categories)
```

and remove the following loop